### PR TITLE
Make info icons less obtrusive

### LINKED ISSUE #40863: Make info hover icons less intrusive
A couple of follow ups to the info icon feature to make them less intrusive.

[Product doc](https://www.notion.so/metabase/Surface-column-info-follow-ups-9ec2c0316c0d4a2b8c6b7e123a027a48)

- [ ] Hide column info icons entirely if there is no description on the column
- [ ] Try adding a delay before displaying the hover and evaluate how that feels, otherwise try out on-click instead of on-hover
- [ ] Remove the icon bg, make the icon blue, and change the font weight from bold to normal
- [ ] This is appearing for custom expressions, where it’s impossible to set a description for them

### PR DESCRIPTION
**Pull request by romeovs**
Closes https://github.com/metabase/metabase/issues/40863

### How to verify

Describe the steps to verify that the changes are working as expected.

1. Browse data -> Sample Dataset -> Orders
2. Hover the column headers, or info icons in the summarize sidebar and filter popover
3. Verify that there is a small delay for them to open initially (hovering sibling icons should be instant after that).
4. Verify that the semantic type label style has been updated

### Demo

Screenshot 2024-04-01 at 14 28 18

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/components/MetadataInfo/ColumnInfoIcon/ColumnInfoIcon.tsx [hunk 1]
```diff
@@ -10,19 +10,28 @@ import {
   QueryColumnInfoPopover,
   TableColumnInfoPopover,
 } from "../ColumnInfoPopover";
-import { PopoverHoverTarget, HoverParent } from "../InfoIcon";
+import { PopoverHoverTarget, HoverParent, defaultDelay } from "../InfoIcon";
 export { HoverParent };
 
+export type QueryColumnInfoIconProps = QueryColumnInfoPopoverProps & {
+  showIfEmpty?: boolean;
```
Comment (kamilmielnik) [inline frontend/src/metabase/components/MetadataInfo/ColumnInfoIcon/ColumnInfoIcon.tsx:17]: This prop is no longer used in app code (only in tests), we can remove it (and the tests that use it).

Same goes for `TableColumnInfoIconProps`

File: frontend/src/metabase/components/MetadataInfo/ColumnInfoIcon/ColumnInfoIcon.tsx [hunk 2]
```diff
@@ -10,19 +10,28 @@ import {
...
   ...props
-}: QueryColumnInfoPopoverProps) {
+}: QueryColumnInfoIconProps) {
   const { query, stageIndex, column } = props;
   const { description = "" } = query
     ? Lib.displayInfo(query, stageIndex, column)
     : {};
 
+  if (!description && !showIfEmpty) {
+    return null;
```
Comment (kamilmielnik) [inline frontend/src/metabase/components/MetadataInfo/ColumnInfoIcon/ColumnInfoIcon.tsx:32]: Suggestions without info icon now have different height than the ones with the icon.
I think they should have the same size.

In the screenshot the "Total" column does not have an icon and the background color has been added with developer tools to highlight the difference in height.


image

Comment (romeovs) [reply frontend/src/metabase/components/MetadataInfo/ColumnInfoIcon/ColumnInfoIcon.tsx:27]: Good catch. I've updated some other components as well.

Comment (kamilmielnik) [reply frontend/src/metabase/components/MetadataInfo/ColumnInfoIcon/ColumnInfoIcon.tsx:27]: Now rows without icons are 24px high, and the rows with icons are 26px high.
Please fix it before merging :pray:

**Review by kamilmielnik — APPROVED** [review]
Looks good overall, I left 1 comment - please address it before merging :pray:

**Pull request merged**